src/cosl/coordinated_workers/coordinator.py

```diff
@@ -452,6 +455,11 @@ def _s3_config(self) -> Dict[str, Any]:
         s3_config["access_key_id"] = s3_data.pop("access-key")
         s3_config["secret_access_key"] = s3_data.pop("secret-key")
         s3_config["bucket_name"] = s3_data.pop("bucket")
+        ca_chain = s3_data.get("tls-ca-chain")
+        if ca_chain:
+            # put the cacert to disk
+            Path(S3_CA_CERT_PATH).write_text(ca_chain[0])  # TODO: is any cert in the chain good?
```

Comment:
Also, we should push it to the workload container not the charm

Reply:
:facepalm: